### DISCUSSION TITLE
Hibernatable Client WebSockets

### DISCUSSION THREAD

**Discussion by helloimalastair** [cloudflare/workerd#1440] [Ideas]
At the moment, the Hibernatable WebSockets API, while appearing client/server agnostic from the types, only functions with Server-side WebSockets(where the DurableObject is the Server). Would it be possible to add support for Client-side WebSockets to hibernate as well? This would be nice for consuming a WebSocket-based API(real-time data of some kind), and reducing the number of times you have to restart a WebSocket due to a DO getting evicted.

**Comment by kentonv** [comment]
I think this "ought to" be supported, that is, it makes sense technically. I don't think this is something we're likely to work on in the near future, though.

**Reply by Rubydesic** [reply]
This feature would be really useful specifically for Discord bots, which require connecting to the Discord Gateway API over WebSockets for full functionality. Cloudflare's docs have a Discord bot example, but it uses the very limited HTTP webhook API and so is not practical for most applications.
